google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1667,15 +1690,45 @@ def signed_url path = nil,
         #   post.fields[:signature] #=> "ABC...XYZ="
         #   post.fields[:policy] #=> "ABC...XYZ="
         #
-        def post_object path, policy: nil, issuer: nil,
-                        client_email: nil, signing_key: nil,
-                        private_key: nil
+        def post_object path,
```

Comment:
@quartzmo I'm thinking this is a bit complex of a method signature to be flipping between v2 and v4.

Could you provide an small example using this update for conformance test: https://github.com/googleapis/conformance-tests/blob/master/storage/v1/v4_signatures.json#L371-L378

Reply:
> I'm thinking this is a bit complex of a method signature to be flipping between v2 and v4.

Should a new method be added then? `Bucket#post_object_v4` ?

Reply:
I'm thinking that may be the best idea right now. @jkwlui is doing the same thing in Node.js. Not all languages have support for it so it will be greenfield for most.

Reply:
Added `Bucket#post_object_v4`, PTAL.

Reply:
Also added requested example to the new method.